### PR TITLE
Explicit parameter binding

### PR DESCRIPTION
**Pull request by KathleenDollard**
Allows differently named parameters and paired options/arguments

### CODE REVIEW

**Comment by KathleenDollard** [comment]
Fixes #952

File: src/System.CommandLine/Binding/ModelBinder.cs [hunk 1]
```diff
@@ -219,13 +212,12 @@ public void UpdateInstance<T>(T instance, BindingContext bindingContext)
                         {
                             if (parameterDescriptor.HasDefaultValue)
                                 boundValue = BoundValue.DefaultForValueDescriptor(parameterDescriptor);
-                            else if (parameterDescriptor.AllowsNull && 
+                            else if (parameterDescriptor.AllowsNull &&
                                 ShouldPassNullToConstructor(constructorDescriptor.Parent, constructorDescriptor))
                                 boundValue = BoundValue.DefaultForType(valueDescriptor);
```
Comment (jonsequitur) [inline src/System.CommandLine/Binding/ModelBinder.cs:217]: It looks like this block should also be part of the extracted `GetBoundValue` method.

File: src/System.CommandLine/Invocation/InvocationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+namespace System.CommandLine.Invocation
+{
+    public static class InvocationExtensions
+    {
+        public static void BindParameter(this ICommandHandler handler, ParameterInfo param, Option option)
+        {
+            // check for nulls
+            if (!(handler is ModelBindingCommandHandler bindingHandler))
+            {
+                throw new InvalidOperationException("Cannot bind to this type of handler");
```
Comment (jonsequitur) [inline src/System.CommandLine/Invocation/InvocationExtensions.cs:12]: Maybe we should just make `ModelBindingCommandHandler` public?

**Comment by KathleenDollard** [comment]
Replaced with #1018

**Pull request closed without merging**